### ISSUE TITLE
log file is used by another process

### ISSUE THREAD

**Issue by RemSann** [serilog/serilog-sinks-file#94]
At runtime im not able to read my Serilog file, It is used by another process error is thrown. File.ReadAllBytes is failing. How do i fix this?

**Comment by bartelink** [comment]
Same way you fix it for any other File being written by .NET - sharing mode. Your other question suggests you are doing `shared: true` - this really is not something that's worthy of logging as a question specific to this Sink of this Logger.

**Comment by RemSann** [comment]
I used below code to read the content of log file.
Stream stream = File.Open(_filepath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
                    StreamReader streamReader = new StreamReader(stream);
                    string str = streamReader.ReadToEnd();

**Comment by RemSann** [comment]
now i want to clear the content of log file after reading the content. How to do this?

**Comment by bartelink** [comment]
I have no idea, but I know I'd:

a) look in the code for this sink to find out how it `Open`s the file given the arguments you passed
b) read the docs about your arguments and the `File.Open` ones
c) get from there to the docs about file sharing modes

My guess is that deleting with `File.Delete` is the most direct way, as long as the access modes leave that open.

Your code above will need to do a `using` and/or a `Close()` in order to release the file correctly (as will the logger - you'll need it to release the file, which you might do with `Log.CloseAndFlush` and/or running it in buffered mode).

These are semi-informed guesses and clues; again, in general you're better off researching these things from first principles than waiting for someone on github though (I'm betting there are a heck of relevant answers on SO).

Another question though: why do you feel you need to delete things - destroying a log can really make troubleshooting way more problematic? I'd consider not doing that and/or doing a rename to shift it away instead (though doing that will mean disk space remains occupied until someone comes to clear it, whereas the rolling scheme normally trims it after some time).

Wild idea: if you're trying to implement some scheme where you want to read a log, do a `File.ReadAllLines` and then look for a marker line injected by doing `Log.Warning("Log Virtually Truncated")`

**Comment by RemSann** [comment]
Basically as i had mentioned in my previous question i want to have backup file with .bak extension. since that is not available with serilog framework, i have written a C# code to backup log files. After reading the content of existing log file and writing to the back up file, i would want to clear and content of this existing file to make it a fresh one. File.Delete cannot be used because it is showing "user by  another process" error. I have also tried calling Dispose(which internally calls close and flush) and then do File.Delete. Even this did not help.

**Comment by bartelink** [comment]
I know, I'm aware of the other thing; my answer partially addresses that - I'm afraid this really isnt the forum; this requires:
a) you to do research (which will help your understanding)
b) supplemented with stackoverflow.com

Yes, eventually someone will come along that has fought this dragon before, but it's a file sharing dragon, not a Serilog.Sinks.File dragon - _you're trying to do a non standard thing with a file_.

**Comment by nblumhardt** [comment]
Sounds like you were able to get unblocked @RemSann - closing as this is a bit out-of-scope, but best of luck!
